scripts/projects.js

```diff
@@ -71,39 +69,30 @@
     var parseData = JSON.parse(dataFromStorage);
     Project.loadAll(parseData);
     projectView.renderProjectPage();
-    projectView.showLanguages();
-    projectView.showLibraries();
-  };
-
-  Project.allLanguages = function() {
-    return Project.allProjects.map(function(p) {
-      return p.languages;
-    }).reduce(function(a, b) {
-      return a.concat(b);
-    },[])
-      .reduce(function(acc, cur){
-        if (acc.indexOf(cur) === -1) {
-          acc.push(cur);
-        }
-        return acc;
-      },[]);
+    projectView.show(Project.getFromProject('languages'), 'li', 'languageList');
```

Comment:
The concept of currying has been brought to my attention, so with enough time I hop eot refactor these four show method calls to one or two lines if I can figure out this new concept.